tensorboard/backend/application_test.py

```diff
@@ -535,6 +559,30 @@ def testNameToPluginMapping(self):
     self.assertEqual('foo', mapping['foo'].plugin_name)
     self.assertEqual('bar', mapping['bar'].plugin_name)
 
+  def testNormalRoute(self):
+    self._test_route('/data/plugin/foo/foo_route', True)
+
+  def testNormalRouteIsNotWildcard(self):
+    self._test_route('/data/plugin/foo/foo_route/bogus', False)
+
+  def testMissingRoute(self):
+    self._test_route('/data/plugin/foo/bogus', False)
+
+  def testEmptyRoute(self):
+    self._test_route('', False)
+
+  def testSlashlessRoute(self):
+    self._test_route('runaway', False)
+
+  def testGoodWildcardRoute(self):
+    self._test_route('/data/plugin/bar/wildcard/ok', True)
+  
+  def testBadWildcardRoute(self):
+    self._test_route('/data/plugin/bar/wildcard/bogus', False)
```

Comment:
If I understand correctly, this test is a bit misleading: it suggests
that `…/wildcard/bogus` and `…/wildcard` are handled in the same way,
but they’re not—the former is handled by the plugin, while the latter
never resolves to a plugin app at all. There’s really nothing “bad”
about the `…/wildcard/bogus` route; the plugin handler just chooses to
reject it with the same status code.

What’s the purpose of having the `…/bogus` test case in addition to the
`…/ok` test case, given that they go through exactly the same codepaths
of the code under test?

Reply:
This was really intended more as documentation than as a test per se (i.e., to show that the plugin may yet reject a request matching a wildcard).  Tried to clarify with naming and comments.

Reply:
Much clearer now; thanks!